docs/security-configuration/proxy.md

```diff
@@ -0,0 +1,213 @@
+
+
+## Example
+
+The following example uses an nginx proxy in front of a three-node Elasticsearch cluster. For simplicity, we use hardcoded values for `x-proxy-user` and `x-proxy-roles`. In a real world example you would set these headers dynamically. The example also includes a commented header for use with the extended proxy.
+
+```
+events {
+  worker_connections  1024;
+}
+
+http {
+
+  upstream elasticsearch {
+    server node1.example.com:9200;
+    server node2.example.com:9200;
+    server node3.example.com:9200;
+    keepalive 15;
+  }
+
+  server {
+    listen       8090;
+    server_name  nginx.example.com;
+
+    location / {
+      proxy_pass https://elasticsearch;
+      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+      proxy_set_header x-proxy-user test;
+      proxy_set_header x-proxy-roles test;
+      #proxy_set_header x-proxy-ext-namespace my-namespace;
+    }
+  }
+
+}
+```
+
+The corresponding minimal `config.yml` looks like:
+
+```yml
+---
+_meta:
+  type: "config"
+  config_version: 2
+
+config:
+  dynamic:
+    http:
+      xff:
+        enabled: true
+        internalProxies: '172.16.0.203' # the nginx proxy
+    authc:
+      proxy_auth_domain:
+        http_enabled: true
+        transport_enabled: true
+        order: 0
+        http_authenticator:
+          type: proxy
+          #type: extended-proxy
+          challenge: false
+          config:
+            user_header: "x-proxy-user"
+            roles_header: "x-proxy-roles"
+            #attr_header_prefix: "x-proxy-ext-"
+        authentication_backend:
+          type: noop
+```
+
+The important part is to enable the `X-Forwarded-For (XFF)` resolution and set the IP(s) of the internal proxies correctly:
+
+```yml
+enabled: true
+internalProxies: '172.16.0.203' # nginx proxy
+```
+
+In this case, `nginx.example.com` runs on `172.16.0.203`, so add this IP to the list of internal proxies. Be sure to set `internalProxies` to the minimum number of IP addresses so that the Security plugin only accepts requests from trusted IPs.
+
+
+## Kibana proxy authentication
+
+To use proxy authentication with Kibana, the most common configuration is to place the proxy in front of Kibana and let Kibana pass the user and role headers to the Security plugin.
+
+In this case, the remote address of the HTTP call is the IP of Kibana, because it sits directly in front of Elasticsearch. Add the IP of Kibana to the list of internal proxies:
+
+```yml
+---
+_meta:
+  type: "config"
+  config_version: 2
+
+config:
+  dynamic:
+    http:
+      xff:
+        enabled: true
+        remoteIpHeader: "x-forwarded-for"
+        internalProxies: '<kibana-ip-address>'
+```
+
+To pass the user and role headers that the authenticating proxy adds from Kibana to the Security plugin, add them to the HTTP header whitelist in `kibana.yml`:
+
+```yml
+elasticsearch.requestHeadersWhitelist: ["securitytenant","Authorization","x-proxy-user","x-proxy-roles"]
```

Comment:
I think you need x-forwarded-for here. Also remove Authorization as users should only be using proxy auth?

Reply:
xff, yep, definitely. Thanks, updated.

My understanding is that you must leave `securitytenant` and `Authorization` as-is, though.